### ISSUE TITLE
Modify a process instance by activating elements and setting local variables to a given scope

### ISSUE THREAD

**Issue by saig0** [camunda/zeebe#9689]
## Description

We want to pass variables when activating an element. The variables should be stored as local variables in the given scope.  Each activation instruction can contain variables and a different scope. 

The scope of the variables is selected by its element id. The element instance of the scope should be the same where the element is activated in. 

The variables should be set before activating the element. If the element is embedded in a subprocess then the variables should be set before activating the scope. The variables may be required to create event subscriptions of the element or of an upper flow scope.

For example:
![image](https://user-images.githubusercontent.com/4305769/177247300-05a383a0-2316-4d1c-a54a-6d2aa046e155.png)

* set the variables by writing a `created` event for each activate instruction
  * set the element instance key of the scope as the scope key of the variables

Out of scope:
* element id of scope doesn't exist

Blocked by 
- #9663

**Comment by korthout** [comment]
Please consider 595579443325792255e158c49545b25f27abdcc5 about combining the logic for global and local variables:
>When we look at local variable instuctions, we should spend some time to
see if we can merge these parts and remove the duplication. But if this
leads to additionally complex code, it's likely better to just duplicate
the logic.
